hormonal-hedonists/TLOA/core/game.py

```diff
@@ -0,0 +1,160 @@
+import random
+
+from typing import List, Optional
+
+from TLOA.core.constants import (
+    Actions, AUDIO_PATH, TICK, NUMBER_OF_LANES, LANE_BOUNDS, SHIP_SPAWN_CHANCE,
+    SHIP_SPAWN_RATE, GOLD_SHIP_CHANCE, LIGHT_SOURCE_POS, LIGHT_FOCUS_OFFSET,
+    MIRROR_CANNON_RANGE
+)
+from TLOA.entities import MirrorCannon, BrownShip, GoldenShip, LightRays
+
+from kivy import Logger
+from kivy.clock import Clock, ClockEvent
+from kivy.core.audio import Sound, SoundLoader
+from kivy.event import EventDispatcher
+from kivy.properties import BooleanProperty, BoundedNumericProperty, NumericProperty
+from kivy.vector import Vector
+
+
+class Game(EventDispatcher):
```

Comment:
I like the decision to go for an `EventDispatcher` subclass here. It makes sense, since the `Game` has quite a few possible events that could happen during play. It really helps with reasoning about the flow of execution in the application and it greatly increases the readability.